lib/bokken/types/pt_mobile.ex

```diff
@@ -0,0 +1,64 @@
+defmodule Bokken.Types.Ecto.PtMobile do
+  @moduledoc """
+  PT Phone number type with validation and formatting for Ecto.
+  """
+
+  use Ecto.Type
+
+  def type, do: :string
+
+  # function to transform external data into a runtime format
+  def cast("+351" <> number) do
+    mobile = format(number) 
+             |> List.to_string()
+
+    if is_nil(Regex.run(~r/9[12356]\d{7}/, mobile)) do
+      {:error, [message: "número PT não válido"]}
+    else
+      {:ok, "+351" <> mobile}
+    end
+  end
+  def cast(_number) do
+    {:error, [message: "número sem indicativo"]}
+  end
+
+
+  # when putting the phone number into the database this function 
+  # transforms the data into a specific format to be stored
+  def dump("+351" <> mobile) do
+    {:ok, "+351" <> mobile}  
+  end
+  def dump(_mobile) do
```

Comment:
```suggestion
  def dump(_mobile), do: :error
```

Try this to make it a oneliner 💪